ecommerce/models.py

```diff
@@ -298,10 +313,13 @@ def another_user_already_redeemed(self, user):
             bool:
                 True if the coupon is not automatic and it has already been redeemed by someone else
         """
-        return not self.is_automatic and RedeemedCoupon.objects.filter(
-            coupon=self,
-            order__status=Order.FULFILLED,
-        ).exclude(order__user=user).exists()
+        return (
+            not Coupon.objects.filter(id=self.id).filter(self.is_automatic_qset()).exists() and
+            RedeemedCoupon.objects.filter(
+                coupon=self,
+                order__status=Order.FULFILLED,
+            ).exclude(order__user=user).exists()
```

Comment:
I am not sure you actually need to make the extra query (`RedeemedCoupon.objects.filter`) given that the first filter is `self`. But It looks neat, so it is up to you.